client/allocrunner/taskrunner/task_runner.go

```diff
@@ -1135,6 +1135,22 @@ func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
 				Options:  interpolatedNetworks[0].DNS.Options,
 			}
 		}
+
+		// merge in the DNS set by any CNI plugins
+		netStatus := tr.allocHookResources.GetAllocNetworkStatus()
+		if netStatus != nil && netStatus.DNS != nil {
```

Comment:
If only our language supported Maybe monads!

Reply:
Less silly: we end up needing to nil-check anyways because the `DNS` value can be `nil` (and as noted below, that's potentially required for correct task driver behavior), so doing an error + nil check vs two nil checks feels like a wash.